src/ILLink.RoslynAnalyzer/XmlAnalyzer.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;using System.Collections.Immutable;
+using System.Linq;
+using ILLink.Shared;
+using Microsoft.CodeAnalysis;using Microsoft.CodeAnalysis.Diagnostics;
+using System.IO;
+using Microsoft.CodeAnalysis.Text;
+using System.Text;
+using System.Xml.Linq;
+
+namespace ILLink.RoslynAnalyzer
+{
+	public abstract class XmlAnalyzer : DiagnosticAnalyzer
```

Comment:
Do we have a plan how to share this between the analyzers?

I guess maybe a better question: Why is this a separate analyzer? So that it can generate its own diagnostics for the problems found inside the XMLs?

Reply:
Yes, I was planning to have each analyzer refer to the same "model" and use it however they need to. I was planning to use a separate XML to generate all diagnostics related to the xml, like malformed xml or not being able to resolve a type. This way we wouldn't generate the same warnings multiple times from each analyzer. Since the model would ideally be shared between the linker and analyzer, I didn't want to report diagnostics when the model is being built, so the model may have bad data that the XmlAnalyzer would report.

Reply:
OK- makes sense. It's going to be a bit tricky as there are grey areas of overlap - "is this problem part of the XML, part of the way we apply the XML to the program?". But that's relatively minor.